project/Build.scala

```diff
@@ -56,6 +56,16 @@ object Build extends sbt.Build {
   val newScalaBinaryVersionsInThisRelease: Set[String] =
     Set()
 
+  val javaVersion: Int = {
+    val v = System.getProperty("java.version")
+    v.substring(0, 3) match {
+      case "1.8" => 8
+      case "1.7" => 7
+      case "1.6" => 6
+      case _     => sys.error(s"Unknown JDK version $v")
+    }
```

Comment:
Would the following be OK too?

```scala
            streams.value.log.warning(
                s"Unknown JDK version $v. Assuming maximal compatibility")
            Int.MaxValue
```

It is kind of hard to decide on something meaningful at usage site if you get a `None`. Thanks to the rock-solid backwards compatibility of the JDK, `Int.MinValue` and `Int.MaxValue` seem to actually be reasonable values here...